lib/Conversion/TorchToMhlo/ViewLikeOps.cpp

```diff
@@ -49,17 +59,33 @@ SmallVector<Value, 4> getDimSizesOfTensor(PatternRewriter &rewriter,
     return {};
   }
 
+  auto dims = toPositiveDims(inpDims, rank);
   SmallVector<Value, 4> dimSizes;
-  dimSizes.reserve(rank);
+  dimSizes.reserve(dims.size());
   auto loc = op->getLoc();
-  for (auto d = 0; d < rank; ++d) {
+  for (auto d : dims) {
     dimSizes.emplace_back(rewriter.create<arith::IndexCastOp>(
         loc, rewriter.getIntegerType(kMhloDimSizeBits),
         rewriter.create<tensor::DimOp>(loc, value, d)));
   }
   return dimSizes;
 }
 
+SmallVector<Value, 4> getDimSizesOfTensor(PatternRewriter &rewriter,
+                                          Operation *op, Value value) {
+  auto valueTy = value.getType().dyn_cast<RankedTensorType>();
+  if (!valueTy) {
+    op->emitOpError("getDimSizesOfTensor(): the input is not a ranked tensor");
```

Comment:
Either rewriter.notifyMatchFailure if this is a limitation of the pattern or the check should be in the verifier of the op.

Could the return type be FailureOr<...> ?